apps/courses/src/routes/courses/_helpers.ts

```diff
@@ -343,3 +344,25 @@ export const useCoursePermissionGate = (user, lessons, page, params) => {
 
   return { page, user, lessons, params };
 };
+
+export const userIsMentor = async (user: firebase.User) => {
+  const db = useFirestore();
+  const dbUserRef = db.collection('users').doc(user.uid);
+  const [dbUser, setDbUser] = useState<User>(null);
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      setDbUser((await dbUserRef.get()).data() as User);
+    };
+
+    if (user && user.uid && !dbUser) fetchUser();
+  }, [user, dbUser, dbUserRef]);
+
+  const isMentor =
```

Comment:
The only missing piece here is that we should verify if the current course is included in `mentorable_courses`